endpoint/retry.go

```diff
@@ -53,3 +57,70 @@ func NewExponentialBackoffPolicy(i, m uint, d time.Duration) RetryPolicy {
 		return time.Duration(p) * d, true
 	}
 }
+
+// NewIndefiniteRetryPolicy is the policy that regulates indefinite amount of
+// retry attempts. Similar to NewExponentialBackoffPolicy, this policy allows a
+// fixed number of immediate attempts and calculates the next delay
+// exponentially. However, contrary to NewExponentialBackoffPolicy, this policy
+// always returns true as the second return value denoting that the message
+// should be retried indefinitely.
+//
+// i is the number of immediate attempts. d is a multiplier for the backoff
+// duration. dmax is the maximum backoff duration. randomize is the
+// randomization factor used to randomize the returned duration within the
+// following range:
+//
+// [ d-(d*randomize), d+(d*randomize) ]
+//
+// If d exceeds dmax, dmax value is used to randomize duration.
+func NewIndefiniteRetryPolicy(i uint, d, dmax time.Duration, randomize float64) RetryPolicy {
+	// lastDelay is the last computed delay returned by this retry policy
+	var lastDelay time.Duration
```

Comment:
This state can't be kept; otherwise the retry delay applied to one message will affect all others. It needs to be stateless.

Reply:
Actually, could this be achieved by passing treating an `m` value of `0` passed to `NewExponentialBackoffPolicy` as meaning infinity?